### PR TITLE
[Fix bug 1168359] A/B testing for several tasks under /contribute/signup/

### PR DESCRIPTION
**Pull request by pmclanahan**
Replaces #3059 and adds tests and fixes discovered issues. Cleans a few things
up a bit as well.

### CODE REVIEW

**Comment by pmclanahan** [comment]
@craigcook here's the new one. Deploy and test this on the demo please.

@jgmize check my tests when you get a min please. r?

File: bedrock/mozorg/templates/mozorg/contribute/contribute-tasks-survey.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{# This Source Code Form is subject to the terms of the Mozilla Public
+ # License, v. 2.0. If a copy of the MPL was not distributed with this
+ # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+{% extends "mozorg/contribute/contribute-base.html" %}
+
+{% block page_title %}{{_('Thank you!')}}{% endblock %}
+
+{% block optimizely %}
+  {% if waffle.switch('contribute-signup-optimizely') and LANG == 'en-US' %}
```
Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/contribute/contribute-tasks-survey.html]: If we put the `LANG == 'en-US'` check first, then that will eliminate the memcache and/or db hit for all non-US traffic.

File: bedrock/mozorg/templates/mozorg/contribute/contribute-tasks.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+{% extends "mozorg/contribute/contribute-base.html" %}
+
+{% block body_id %}inquiry{% endblock %}
+
+{% block page_title %}{{_('Get Involved: Volunteer Opportunities at Mozilla')}}{% endblock %}
+
+{% block optimizely %}
+  {% if waffle.switch('contribute-signup-optimizely') and LANG == 'en-US' %}
```
Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/contribute/contribute-tasks.html]: Same here, please put the LANG check first.

File: bedrock/mozorg/templates/mozorg/contribute/signup.html [hunk 3]
```diff
@@ -10,6 +10,12 @@
 
 {% block page_title %}{{_('Get Involved: Volunteer Opportunities at Mozilla')}}{% endblock %}
 
+{% block optimizely %}
+  {% if waffle.switch('contribute-signup-optimizely') and LANG == 'en-US' %}
```
Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/contribute/signup.html]: Same performance nit here

File: bedrock/mozorg/views.py [hunk 4]
```diff
@@ -154,12 +154,68 @@ def get_success_url(self):
...
+            'interest_id': 'dontknow',
+            'lang': form.locale,
+            'source_url': self.request.build_absolute_uri(),
+        }
+
+    def form_valid(self, form):
+        try:
+            basket.request('post', 'get-involved', self.get_basket_data(form))
+        except basket.BasketException as e:
+            if e.code == basket.errors.BASKET_INVALID_EMAIL:
```
Comment (jgmize) [inline bedrock/mozorg/views.py]: NIT: I didn't see a test for this branch

**Comment by jgmize** [comment]
@pmclanahan other than a few nits this LGTM, r+wc.

File: bedrock/mozorg/views.py (hunk 4 above)
Comment (pmclanahan) [inline bedrock/mozorg/views.py]: Yeah. Was late getting this in. Didn't find it a big deal. Can add one though if you want.

**Comment by pmclanahan** [comment]
Switched conditionals and added a test for the email error branch.

**Comment by pmclanahan** [comment]
@craigcook after these tests pass I believe we're good to merge. You good with that as far as tests on demo go?

**Comment by craigcook** [comment]
r+ on the front end (though I may make some wee tweaks and update some copy in a followup PR)

**Pull request merged**